MolecularNodes/mda.py

```diff
@@ -0,0 +1,409 @@
+                "value": att_vdw_radii,
+                "type": "FLOAT",
+                "domain": "POINT",
+            },
+            {"name": "res_id", "value": att_res_id, "type": "INT", "domain": "POINT"},
+            {
+                "name": "res_name",
+                "value": att_res_name,
+                "type": "INT",
+                "domain": "POINT",
+            },
+            {
+                "name": "b_factor",
+                "value": att_b_factor,
+                "type": "FLOAT",
+                "domain": "POINT",
+            },
+            {
+                "name": "chain_id",
+                "value": att_chain_id,
+                "type": "INT",
+                "domain": "POINT",
+            },
+            {
+                "name": "atom_types",
+                "value": att_atom_type,
+                "type": "INT",
+                "domain": "POINT",
+            },
+            {
+                "name": "is_backbone",
+                "value": att_is_backbone,
+                "type": "BOOLEAN",
+                "domain": "POINT",
+            },
+            {
+                "name": "is_alpha_carbon",
+                "value": att_is_alpha_carbon,
+                "type": "BOOLEAN",
+                "domain": "POINT",
+            },
+            {
+                "name": "is_solvent",
+                "value": att_is_solvent,
+                "type": "BOOLEAN",
+                "domain": "POINT",
+            },
+            {
+                "name": "is_nucleic",
+                "value": att_is_nucleic,
+                "type": "BOOLEAN",
+                "domain": "POINT",
+            },
+            {
+                "name": "is_peptide",
+                "value": att_is_peptide,
+                "type": "BOOLEAN",
+                "domain": "POINT",
+            },
+        )
+
+        for att in attributes:
+            # tries to add the attribute to the mesh by calling the 'value' function which returns
+            # the required values do be added to the domain.
+            try:
+                obj.add_attribute(
+                    mol_object, att["name"], att["value"](), att["type"], att["domain"]
+                )
+            except:
+                warnings.warn(f"Unable to add attribute: {att['name']}.")
+
+        self.ags[mol_object.name] = ag
+        self.ag_names.append(mol_object.name)
+
+        nodes.create_starting_node_tree(
+            obj=mol_object, starting_style=bpy.context.scene.mol_import_default_style
+        )
+
+    @persistent
+    def update_trajectory(self, frame):
+        # TODO: how about a dynamic selection with variable number of atoms?
+        self.trajectory[frame]
+        for name in self.ag_names:
+            ag = self.ags[name]
+            mol_object = bpy.data.objects[name]
+            locations = ag.positions * self.world_scale
+            for vert, loc in zip(mol_object.data.vertices, locations):
+                vert.co = loc
```

Comment:
For updating the locations, you can write directly to the attributes of the mesh from numpy arrays instead of having to loop through the vertices individually. 

In my testing, I got approximately ~100x speed up by changing these two lines to the following: 
```python
for vert, loc in zip(mol_object.data.vertices, locations):
    vert.co = loc
```

```python
mol_object.data.attributes['position'].data.foreach_set('vector', locations.reshape(-1))
mol_object.data.update()
```

The only _gotcha_ is that currently to write to vector attributes such as position, you have to supply a 1D array so we have to just reshape it before setting. Not doing this can segfault blender. They have plans to improve this API eventually but this is what we currently have to work with.